gateway/handler.go

```diff
@@ -277,6 +277,19 @@ func (i *handler) getOrHeadHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if !i.isTrustedMode(r) {
+		switch responseFormat {
+		case "application/vnd.ipld.raw",
```

Comment:
Flagging that IMO for trustless mode we should error if the request is not for `/ipfs/cid` (i.e. any IPNS or path elements require trust)